src/probnum/random_variables/_normal.py

```diff
@@ -174,29 +182,30 @@ def __init__(
             var = self._dense_var
             entropy = self._dense_entropy
 
-            if cov_cholesky is None:
-                self._compute_cov_cholesky = self.dense_cov_cholesky
-            else:
-                if not isinstance(cov_cholesky, type(self._cov)):
+            self._compute_cov_cholesky = self.dense_cov_cholesky
+
+            # Ensure that the Cholesky factor has the same type as the covariance,
+            # and, if necessary, promote data types. Check for (in this order): type, shape, dtype.
+            if self._cov_cholesky is not None:
```

Comment:
took me a bit to get this code block, so I thought I'd document what I learned ;)